src/Debug.c

```diff
@@ -47,37 +47,29 @@ void Ros_Debug_BroadcastMsg(char* fmt, ...)
 
     if (ros_DebugSocket == -1)
         Ros_Debug_Init();
-    
-   // Timestamp
 
-    time_t DEBUG_MSG_TIMESTAMP;
-    struct tm ts;
-    struct timeval tv;
-
-
-    time(&DEBUG_MSG_TIMESTAMP);
-
-    localtime_r(&DEBUG_MSG_TIMESTAMP, &ts);
-
-    strftime(Formatted_Time, sizeof(Formatted_Time), "%a %Y-%m-%d %H:%M:%S", &ts);
-
-    gettimeofday(&tv, NULL);
-    long msec = tv.tv_usec / 1000;
-
-    snprintf(Formatted_Time, sizeof(Formatted_Time), "%s.%03ld", Formatted_Time, msec);
-
-    size_t Timestamp_Length = strlen(Formatted_Time);
-    size_t Debug_Message_Length = strlen(str);
-
-    // Pre-pending the timestamp to the debug message 
-
-    if (Timestamp_Length + Debug_Message_Length + 1 < MAX_DEBUG_MESSAGE_SIZE)
+    // Timestamp
+    struct tm synced_time;
+    int64_t nanosecs = rmw_uros_epoch_nanos();
+    builtin_interfaces__msg__Time debug_msg_timestamp;
+
+    Ros_Nanos_To_Time_Msg(nanosecs, &debug_msg_timestamp);
+    time(&debug_msg_timestamp);
```

Comment:
IIUC, the `time` function will put the current time into `debug_msg_timestamp`. But, this is not needed because the current timestamp was already obtained above with `rmw_uros_epoch_nanos`. I think that this line can be deleted.

Reply:
I did try by removing that line but what is happening is that for the fist few messages, before the message "Attempting to connect to the micro-ROS agent", the timestamp being showed is the epoch time (THU 1970-01-01 00:00:00.000) and it is only after this message that it shows the current timestamp.

Reply:
I see. Then I would suggest making the time source conditional on whether the agent is connected.

```
if (g_Ros_Communication_AgentIsConnected)
{
    //get synchronized time from the agent
    int64_t nanosecs = rmw_uros_epoch_nanos();
    Ros_Nanos_To_Time_Msg(nanosecs, &debug_msg_timestamp);
}
else
{
    //rmw_uros_epoch_nanos cannot sync with agent because it's not connected
    time(&debug_msg_timestamp);
}
```

Reply:
Okay, I will replace the time function with the conditional

Reply:
I'm not sure you can pass an instance of `builtin_interfaces__msg__Time` to `time(..)`.